src/Stratis.Bitcoin.Features.Consensus/ConsensusErrors.cs

```diff
@@ -3,181 +3,177 @@
+    /// A consensus error used to specify different types of reasons a block does not confirm to the consensus rules.
+    /// </summary>
     public class ConsensusError
     {
+        /// <summary>
+        /// The code representing this consensus error.
+        /// </summary>
         public string Code { get; }
 
+        /// <summary>
+        /// A user friendly message to describe this error.
+        /// </summary>
         public string Message { get; }
 
+        /// <summary>
+        /// A method that will throw a <see cref="ConsensusErrorException"/> with the current consensus error.
+        /// </summary>
         public void Throw()
         {
             throw new ConsensusErrorException(this);
         }
 
+        /// <summary>
+        /// Initialize a new instance of <see cref="ConsensusErrorException"/>.
+        /// </summary>
+        /// <param name="code">The error code that represents the current consensus breaking error.</param>
+        /// <param name="message">A user friendly message to describe this error.</param>
         public ConsensusError(string code, string message)
         {
             Guard.NotEmpty(code, nameof(code));
             Guard.NotEmpty(message, nameof(message));
-
+            
             this.Code = code;
             this.Message = message;
         }
 
+        /// <inheritdoc />
         public override bool Equals(object obj)
         {
             ConsensusError item = obj as ConsensusError;
-            if (item == null)
-                return false;
-            return this.Code.Equals(item.Code);
+
+            return (item != null) && (this.Code.Equals(item.Code));
         }
 
+        /// <summary>
+        /// Compare two instances of <see cref="ConsensusError"/> are the same.
+        /// </summary>
+        /// <param name="a">first instance to compare.</param>
+        /// <param name="b">Second instance to compare.</param>
+        /// <returns><c>true</c> if bother instances are the same.</returns>
         public static bool operator ==(ConsensusError a, ConsensusError b)
         {
-            if (ReferenceEquals(a, b))
+            if (Object.ReferenceEquals(a, b))
                 return true;
+
             if (((object)a == null) || ((object)b == null))
                 return false;
+
             return a.Code == b.Code;
         }
 
+        /// <summary>
+        /// Compare two instances of <see cref="ConsensusError"/> are not the same.
+        /// </summary>
         public static bool operator !=(ConsensusError a, ConsensusError b)
         {
             return !(a == b);
         }
-
+        
+        /// <inheritdoc />
         public override int GetHashCode()
         {
             return this.Code.GetHashCode();
         }
 
+        /// <inheritdoc />
         public override string ToString()
         {
-            return this.Code + ": " + this.Message;
+            return string.Format("{0} : {1}", this.Code, this.Message);
```

Comment:
`+` is faster than string.Format ... we've discussed this already :)

Reply:
and also before it was `{0}: {1}` now you introduced ugly space before `:`, use the old version here anyway

Reply:
`+` is faster? we did not discuss concatenation this only interpolation.

Reply:
yes, + is faster, but logging is different case

in logging string.Format is faster because if the logging is not at that level the string is not formatted, whereas if we used + it would be processed

so more exactly ... + would be faster even in logging, but only if the logging actually happened. if logging did not happen, string.Format is faster, which is what we care about in logging

in ToString() methods, + is faster

Reply:
ok fair enough but this is not a critical path so I am fine with keeping string.Format in this case.

Reply:
i know, it is just another best practice
so i'd also be happy for this to be sF, but why i did mention this and actually wanted the change is because you changed it. you just shouldn't touch it :) 
so i mean, if you touch it, improve it, not make it worse :-)